core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -5445,29 +5455,30 @@ targetNamespace, castNonNull(catalogReader.unwrap(Prepare.CatalogReader.class)),
       throw newValidationError(createTable, RESOURCE.createTableRequiresAsQuery());
     }
 
-    final SqlValidatorScope queryScope = requireNonNull(scopes.get(queryNode));
-
     //Note, this can either a row expression or a query expression with an optional ORDER BY
-    if (queryNode instanceof SqlSelect) {
-      final SqlSelect sqlSelect = (SqlSelect) queryNode;
-      validateSelect(sqlSelect, unknownType);
-    } else {
-      validateQuery(queryNode, queryScope, unknownType);
-    }
-
-    final SqlValidatorNamespace queryNS = getNamespaceOrThrow(queryNode);
-    final DdlNamespace createTableNS = (DdlNamespace) getNamespaceOrThrow(createTable);
+    //We're not currently handling the row expression case.
+
+    SqlValidatorScope createTableScope = this.getCreateTableScope(createTable);
+    // In order to be sufficiently general to the input of Create table,
+    // We have to validate this expression in the overall scope of the create table node,
+    // Since the associated query node
+    // doesn't necessarily have to be a select (notably, validaSelect doesn't work if the query has
+    // any 'with' clauses)
+    // Note that we also can't use validateScopedExpression, as this can rewrite the sqlNode
+    // via a call to performUnconditionalRewrites, which should have already been done by the time
+    // that we reach this points, and calling performUnconditionalRewrites twice is likely invalid.
+    queryNode.validate(this, createTableScope);
+    // (Note: for create table LIKE (WIP) if queryNode is identifier, we may need additional work to
```

Comment:
add issue link for create table LIKE (WIP)

Reply:
done